protostar/upgrader/upgrade_info_writer_thread.py

```diff
@@ -0,0 +1,44 @@
+from threading import Thread
+
+# pylint: disable=redefined-builtin
+from requests.exceptions import ConnectionError
+
+from protostar.upgrader.upgrade_remote_checker import UpgradeRemoteChecker
+from protostar.upgrader.upgrade_toml import UpgradeTOML
+
+
+class UpgradeInfoWriterThread:
```

Comment:
I wonder if this class wouldn't be simpler are more robust if it would check remote. asynchronously in the background (possibly using asyncio features rather than daemon thread), and then perform file write in `__exit__` method, **synchronously** when program ends.

Generally this looks like a thing to be possibly written using `@asynccontextmanager`.

Reply:
It could. I'll take a look on contextmanager stuff because I didn't use it before.

Reply:
I tried to use asyncio tasks without success. I end up leaving thread hence I left it as a class and didn't use `@asynccontextmanager`.

The task created by the `create_task` didn't start. I think the event loop on the main thread was busy, and the task never had a chance to run.